packages/playwright/src/test.ts

```diff
@@ -0,0 +1,239 @@
+        ...results.violations,
+        ...results.incomplete
+      ];
+      assert.strictEqual(all.length, 1);
+      assert.strictEqual(all[0].id, 'region');
+    });
+
+    it('only runs the provided rules as a string', async () => {
+      await page.goto(`${addr}/index.html`);
+      const results = await new AxeBuilder({ page })
+        .withRules('region')
+        .analyze();
+      const all = [
+        ...results.passes,
+        ...results.inapplicable,
+        ...results.violations,
+        ...results.incomplete
+      ];
+      assert.strictEqual(all.length, 1);
+      assert.strictEqual(all[0].id, 'region');
+    });
+  });
+
+  describe('options', () => {
+    it('passes options to axe-core', async () => {
+      await page.goto(`${addr}/index.html`);
+      const results = await new AxeBuilder({ page })
+        .options({ rules: { region: { enabled: false } } })
+        .analyze();
+      const all = [
+        ...results.passes,
+        ...results.inapplicable,
+        ...results.violations,
+        ...results.incomplete
+      ];
+      assert.isTrue(!all.find(r => r.id === 'region'));
+    });
+  });
+
+  describe('withTags', () => {
+    it('only rules rules with the given tag(s) as an array', async () => {
+      await page.goto(`${addr}/index.html`);
+      const results = await new AxeBuilder({ page })
+        .withTags(['best-practice'])
+        .analyze();
+      const all = [
+        ...results.passes,
+        ...results.inapplicable,
+        ...results.violations,
+        ...results.incomplete
+      ];
+      assert.isOk(all);
+      for (const rule of all) {
+        assert.include(rule.tags, 'best-practice');
+      }
+    });
+
+    it('only rules rules with the given tag(s) as a string', async () => {
+      await page.goto(`${addr}/index.html`);
+      const results = await new AxeBuilder({ page })
+        .withTags('best-practice')
+        .analyze();
+      const all = [
+        ...results.passes,
+        ...results.inapplicable,
+        ...results.violations,
+        ...results.incomplete
+      ];
+      assert.isOk(all);
+      for (const rule of all) {
+        assert.include(rule.tags, 'best-practice');
+      }
+    });
+
+    it('No results provided when the given tag(s) is invalid', async () => {
+      await page.goto(`${addr}/index.html`);
+      const results = await new AxeBuilder({ page })
+        .withTags(['foobar'])
+        .analyze();
+
+      const all = [
+        ...results.passes,
+        ...results.inapplicable,
+        ...results.violations,
+        ...results.incomplete
+      ];
+      // Ensure all run rules had the "foobar" tag
+      assert.deepStrictEqual(0, all.length);
+    });
+  });
+
+  describe('iframe tests', () => {
+    it('injects into nested frames', async () => {
+      await page.goto(`${addr}/nested-frames.html`);
+      const results = await new AxeBuilder({ page }).analyze();
+      assert.strictEqual(results.incomplete.length, 0);
```

Comment:
This test would be more clear on its purpose and more resilient if the test looked for the `frame-tested` rule in the `passes` results. 

As it stands it was confusing why testing for incomplete would prove it was injected into frames since removing the iframe from the fixture would still pass the test. If there was no iframe on the page the `frame-tested` rule would move to `inapplicable` results